### PR TITLE
AM2R: long beam logic adjustments

### PR DESCRIPTION
**Pull request by Miepee**

### CODE REVIEW

File: randovania/games/am2r/logic_database/GFS Thoth.txt [hunk 1]
```diff
@@ -268,8 +268,9 @@ Extra - light_level: 3
           Combat (Advanced) or Normal Damage ≥ 399
           Any of the following:
               # Item requirements
-              Super Missiles ≥ 7 or Combat (Intermediate)
+              Super Missiles ≥ 7 or Combat (Advanced)
               Morph Ball and Power Bombs ≥ 10
+              Long Beam and Combat (Intermediate)
```
Comment (JeffGainsNGames) [inline randovania/games/am2r/logic_database/GFS Thoth.txt:273]: Genesis with only long beam is intermediate? How many regular shots does it take? unless this is supposed to be in the `all`?

File: randovania/games/am2r/logic_database/Industrial Complex.txt [hunk 2]
```diff
@@ -473,8 +473,9 @@ Extra - light_level: 0
       All of the following:
           Any of the following:
               # Weapon Requirements
-              Missiles ≥ 50 or Spazer Beam or Super Missiles ≥ 6 or Wave Beam or Combat (Intermediate)
+              Missiles ≥ 50 or Spazer Beam or Super Missiles ≥ 6 or Wave Beam or Combat (Advanced)
               Power Bombs ≥ 4 and Can Use Power Bombs
+              Long Beam and Combat (Intermediate)
```
Comment (JeffGainsNGames) [inline randovania/games/am2r/logic_database/Industrial Complex.txt:478]: so on intermediate you are expected to fight Torizo with just long beam for weapons? Seems difficult
unless this is supposed to go in the `all`

File: randovania/games/am2r/logic_database/header.txt [hunk 3]
```diff
@@ -228,7 +228,7 @@ Templates
               Missiles ≥ 30 or Super Missiles ≥ 6
               All of the following:
                   # Missile-Less Requirements
-                  Charge Beam
+                  Charge Beam and Underwater Power Grip Wall
```
Comment (JeffGainsNGames) [inline randovania/games/am2r/logic_database/header.txt:231]: Personally I would rather this be on the connection to the dual gamma's for readability and leave this as fight requirements but I won't block the PR over it.

File: randovania/games/am2r/logic_database/GFS Thoth.txt (hunk 1 above)
Comment (Miepee) [reply randovania/games/am2r/logic_database/GFS Thoth.txt:273]: this is just what the previous connection was, didn't change it yet. It should probably be higher, dunno how man shots it takes.

File: randovania/games/am2r/logic_database/Industrial Complex.txt (hunk 2 above)
Comment (Miepee) [reply randovania/games/am2r/logic_database/Industrial Complex.txt:478]: see genesis comment.

File: randovania/games/am2r/logic_database/header.txt (hunk 3 above)
Comment (Miepee) [reply randovania/games/am2r/logic_database/header.txt:231]: IMO it fits nicer here, because you have to use those abilities during the combat.

**Pull request merged**